src/js.rs

```diff
@@ -978,6 +978,114 @@ extern "C" {
+    pub fn construct(target: &Function, arguments_list: &Array) -> Result<JsValue, JsValue>;
+    #[wasm_bindgen(static_method_of = Reflect, js_name = construct, catch)]
+    pub fn construct_with_new_target(target: &Function, arguments_list: &Array, new_target: &Function) -> Result<JsValue, JsValue>;
+
+    /// The static Reflect.defineProperty() method is like Object.defineProperty() 
+    /// but returns a Boolean.
+    /// 
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/defineProperty
+    #[wasm_bindgen(static_method_of = Reflect, js_name = defineProperty, catch)]
+    pub fn define_property(target: &Object, property_key: &JsValue, attributes: &Object) -> Result<JsValue, JsValue>;
+
+    /// The static Reflect.deleteProperty() method allows to delete properties. 
+    /// It is like the delete operator as a function.
+    /// 
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/deleteProperty
+    #[wasm_bindgen(static_method_of = Reflect, js_name = deleteProperty, catch)]
+    pub fn delete_property(target: &JsValue, key: &JsValue) -> Result<JsValue, JsValue>;
+
+    /// The static Reflect.get() method works like getting a property from 
+    /// an object (target[propertyKey]) as a function.
+    /// 
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/get
+    #[wasm_bindgen(static_method_of = Reflect, catch)]
+    pub fn get(target: &JsValue, key: &JsValue) -> Result<JsValue, JsValue>;
+
+    /// The static Reflect.getOwnPropertyDescriptor() method is similar to 
+    /// Object.getOwnPropertyDescriptor(). It returns a property descriptor 
+    /// of the given property if it exists on the object, undefined otherwise.
+    /// 
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/getOwnPropertyDescriptor
+    #[wasm_bindgen(static_method_of = Reflect, js_name = getOwnPropertyDescriptor, catch)]
+    pub fn get_own_property_descriptor(target: &JsValue, property_key: &JsValue) -> Result<JsValue, JsValue>;
+
+    /// The static Reflect.getPrototypeOf() method is almost the same 
+    /// method as Object.getPrototypeOf(). It returns the prototype 
+    /// (i.e. the value of the internal [[Prototype]] property) of 
+    /// the specified object.
+    /// 
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/getPrototypeOf
+    #[wasm_bindgen(static_method_of = Reflect, js_name = getPrototypeOf, catch)]
+    pub fn get_prototype_of(target: &JsValue) -> Result<JsValue, JsValue>;
+
+    /// The static Reflect.has() method works like the in operator as a function.
+    /// 
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/has
+    #[wasm_bindgen(static_method_of = Reflect, catch)]
+    pub fn has(target: &JsValue, property_key: &JsValue) -> Result<JsValue, JsValue>;
```

Comment:
I think that this signature is incorrect. The `target` should rather be of `&Object` type, thus eliminating a need for `catch` (`TypeError` is only thrown when `target` is not an `Object` - but here we avoid that problem at type-checking level).

Could you please review other methods and apply the change (where needed)?

Reply:
I checked all other functions and this applies to all of them. Thanks for clarifying!